plugin.yaml

```diff
@@ -29,26 +28,26 @@ dsl_definitions:
       type: string
       default: ''
 
-    kwargs: &data_type_kwargs
-      kwargs:
-        description: A dictionary of keys and values that is not validated but will override other values in the resource config.
-        default: {}
+  kwargs: &data_type_kwargs
+    kwargs:
+      description: A dictionary of keys and values that is not validated but will override other values in the resource config.
+      default: {}
 
-    name: &data_type_name
-      name:
-        description: Resource name.
-        type: string
-        required: true
+  name: &data_type_name
+    name:
+      description: Resource name.
+      type: string
+      required: false
```

Comment:
@EarthmanT Some of the resources are marked as required name value, cannot we override it for each node type ?

Reply:
I want to write a `create_name` utility method that will generate a name if none is given, and then to add it to the `create` method for those resources that require it.

Reply:
Make sense